src/intents/DSL/schema.ts

```diff
@@ -0,0 +1,111 @@
+import { TokenStandard } from "@brinkninja/types";
+import Joi from "joi";
+import { joi } from "../../internal/joiExtended";
+
+// Object.keys returns keys and values that's why we're filtering out the values using isNaN
+const tokenStandards = Object.keys(TokenStandard).filter(key => isNaN(Number(key)));
+
+export const TokenArgsSchema = joi.alternatives().try(
+  joi.ethereumAddress(),
+  joi.object({
+    address: joi.ethereumAddress().required(),
+    standard: joi.string().valid(...tokenStandards).optional(),
+    idsMerkleRoot: joi.string().optional(),
+    id: joi.bigIntish().optional(),
+    disallowFlagged: joi.boolean().optional(),
+  })
+);
+
+
+export const replaySchema = joi.object({
+  nonce: joi.bigIntish().min(1).required(),
+  runs: joi.string().valid('ONCE', 'UNTIL_CANCELLED').required(),
+});
+
+export const intervalConditionSchema = joi.object({
+  type: joi.string().valid('interval').required(),
+  id: joi.bigIntish().required(),
+  interval: joi.uint().required(),
+  startBlock: joi.uint().optional(),
+  maxIntervals: joi.uint().optional(),
+});
+
+export const blockConditionSchema = joi.object({
+  type: joi.string().valid('block').required(),
+  blockNumber: joi.uint().required(),
+  state: joi.string().valid('MINED', 'NOT_MINED').required()
+});
+
+export const nonceConditionSchema = joi.object({
+  type: joi.string().valid('nonce').required(),
+  nonce: joi.bigIntish().min(1).required(),
+  state: joi.string().valid('USED', 'NOT_USED').required(),
+});
+
+export const priceConditionSchema = joi.object({
+  type: joi.string().valid('price').required(),
+  price: joi.uint().required(),
+  operator: joi.string().valid('lt', 'gt').required(),
+  tokenA: TokenArgsSchema.required(),
+  tokenB: TokenArgsSchema.required(),
+  twapInterval: joi.uint().optional(),
+  twapFeePool: joi.number().integer().valid(500, 3000, 10000).optional(),
```

Comment:
could we get these values from type enum?

Reply:
Ok there's this weird thing where getting enum values get transformed to `string`. and i don't know how to keep them as `numbers`.

Reply:
but in any case, brink-types do not define the possible values for this. at least for now.